tests/Doctrine/Tests/DBAL/ConnectionTest.php

```diff
@@ -701,7 +714,7 @@ public static function dataCallConnectOnce(): iterable
     public function testCallConnectOnce(string $method, array $params): void
     {
         $driverMock   = $this->createMock(Driver::class);
-        $pdoMock      = $this->createMock(Connection::class);
+        $pdoMock      = $this->createMock(PDO::class);
```

Comment:
Same as above, it likely doesn't have to be a PDO connection.

Reply:
I don't know how to change this, if it needs to be changed. The test is mocking `connect` and `connect` is the only method that sets `$_conn`, other than passing in a PDO object to the constructor. Which, now that it has the instanceof check, must be a PDO object.

I was able to change the other tests, just not this one.

Reply:
See the implementation from `3.x`: https://github.com/doctrine/dbal/blob/bd5fc0d72861caeed7aaef91756d3e35527aab67/tests/ConnectionTest.php#L587-L601